openssl-dynamic/src/main/c/ssl.c

```diff
@@ -725,6 +781,19 @@ TCN_IMPLEMENT_CALL(jint, SSL, initialize)(TCN_STDARGS, jstring engine)
             if (ee)
                 ENGINE_free(ee);
         }
+        if (ee != NULL) {
```

Comment:
This check seem strange, since `ee` will have been freed and it's possible for `ee != NULL` but `err == APR_ENOTIMPL`. Should this block be moved into the previous block?